### PR TITLE
apiserver cluster egress metrics: backport to release-1.25

### PR DESCRIPTION
**Pull request by jkh52**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

1. Backport apiserver metric related to cluster egress, to improve observability.

apiserver_egress_dialer_dial_start_total

2. Register  (related) new metrics added in konnectivity-client https://github.com/kubernetes/kubernetes/pull/114929

#### Special notes for your reviewer:

UPDATED on 2023-01-27

* All merge conflicts were imports only.
* These commits are present in master branch.
* This now includes the fix #115327

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by Verolop** [comment]
@jkh52 @liggitt  can we get this lgtm'd and approved, please? 
thanks!

**Comment by jkh52** [comment]
Can a reviewer approve this?

Motivation: The working group at https://github.com/kubernetes-sigs/apiserver-network-proxy has been steadily fixing bugs. Backporting these metrics to earlier minor versions improves observability and helps inform decision to take those ANP fixes.

Risk: this seems minimal beyond the few new metrics (which have appropriately low cardinality, etc).

**Comment by Verolop** [comment]
@jkh52 got it.  We typically only approve backports when they are ready for all the relevant branches, but this makes sense. Thanks for the background.

**Comment by jkh52** [comment]
@logicalhan could you approve this, since you already did for release-1.24? (https://github.com/kubernetes/kubernetes/pull/115561)

**Review by logicalhan — APPROVED** [review]
/lgtm
/approve

**Comment by jkh52** [comment]
friendly ping.

/assign @Verolop

**Comment by liggitt** [comment]
I have a hard time seeing this fall within the guidelines for cherry-picks (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-release/cherry-picks.md#what-kind-of-prs-are-good-for-cherry-picks)

It's not a critical bug fix, and it's not an obviously safe change – expanding an interface, some tricky wiring/plumbing of metrics initialization (which actually bit us [in master](https://github.com/kubernetes/kubernetes/issues/115326)), and registering additional metrics by default

**Pull request closed without merging**
